### PR TITLE
compactor: fix TestPeriodic

### PR DESCRIPTION
**Pull request by xiang90**
compactor: fix TestPeriodic

Perviously, we advance checkCompactionInterval more than we should.
The compaction might happen nondeterministically since there is no
synchronization before we call clock.Advance().

The number of rg.Wait() should be equal to the number of Advance() if
compactor routin and test routin run at the same pace. However, in our current
test, we call Advance() more than rg.Wait().

It works OK when the compactor routine runs "slower" than the test routine, which
is the common case. However, when the speed changes, the compactor ruotine might
block rg.Rev() since there is not enough calls of rg.Wait().

This commit force the compactor and test routin runs at the same pace. And we supply
the exact number of Advance() and wg.Wait() that compactor needs.

/cc @heyitsanthony

### CODE REVIEW

File: compactor/compactor_test.go [hunk 1]
```diff
@@ -39,16 +39,22 @@ func TestPeriodic(t *testing.T) {
 	tb.Run()
 	defer tb.Stop()
 
+	time.Sleep(time.Second * 5)
```
Comment (heyitsanthony) [inline compactor/compactor_test.go]: left over from debugging?

Comment (xiang90) [inline compactor/compactor_test.go]: removed.

**Comment by heyitsanthony** [comment]
lgtm. Thanks for cracking this one

**Comment by fanminshi** [comment]
`s/routin/routine` in the commit msg?

**Comment by xiang90** [comment]
@fanminshi fixed.

**Pull request merged**